src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http2.cs

```diff
@@ -1598,19 +1598,36 @@ public async Task Http2GetAsync_TrailigPseudo_Throw()
         }
 
         [Fact]
-        [ActiveIssue(38708)]
         public async Task InboundWindowSize_Exceeded_Throw()
         {
             var semaphore = new SemaphoreSlim(0);
 
             await Http2LoopbackServer.CreateClientAndServerAsync(
                 async uri =>
                 {
-                    using HttpClient client = CreateHttpClient();
-                    using HttpResponseMessage response = await client.GetAsync(uri, HttpCompletionOption.ResponseHeadersRead);
+                    // An exception will be thrown by either GetAsync or ReadAsStringAsync once
+                    // the inbound window size has been exceeded. Which one depends on how quickly
+                    // ProcessIncomingFramesAsync() can read data off the socket.
+                    Exception requestException = await Assert.ThrowsAsync<HttpRequestException>(async () =>
+                    {
+                        using HttpClient client = CreateHttpClient();
+                        using HttpResponseMessage response = await client.GetAsync(uri, HttpCompletionOption.ResponseHeadersRead);
+
+                        // Keep client open until server is done.
+                        await semaphore.WaitAsync(10000);
+
+                        await response.Content.ReadAsStringAsync();
+                    });
+
+                    // A Http2ProtocolException will be present somewhere in the inner exceptions.
+                    // Its location depends on which method threw the exception.
+                    while (requestException?.GetType().FullName.Equals("System.Net.Http.Http2ProtocolException") == false)
```

Comment:
do we need to be that specific about internal exceptions? since it's not user facing it might be worth checking just outer/public part or check if implements some public base Exception type

Reply:
The only change I can make here is to scan `Message` for `FLOW_CONTROL_ERROR` directly in the loop instead of looking for a specific exception first. I'm not really supper happy with either, but the current way is at least a little more explicit.

Another option is to make `Http2ProtocolException` and `Http2ProtocolErrorCode` public, but I'm not convinced it's useful for users to have access to those.

Reply:
>Another option is to make Http2ProtocolException and Http2ProtocolErrorCode public, but I'm not convinced it's useful for users to have access to those.

We discussed these a while ago. We decided not to make these public yet.  We may reconsider after .NET Core 3.0.  And to do so requires an API review as well.